src/data_conn.c

```diff
@@ -600,13 +603,21 @@ replica_thread(void *arg)
 			if (ret == -1)
 				goto exit;
 		}
-		/* Need to handle wait IOs that are taking more than IO timeout */
-		// handle_data_conn_error(r);
-		// return -1;
+
+		/* check for replica IO timeout */
+		pending_cmd = TAILQ_FIRST(&r->waitq);
```

Comment:
we need to be looking at HEAD of all queues..

Reply:
ok